tests/runner/src/lib.rs

```diff
@@ -138,10 +143,15 @@ where
         )
     });
 
-    let exit_status = child.wait().unwrap();
+    let Some(exit_status) = child.wait_timeout(Duration::new(120, 0)).unwrap() else {
+        child
+            .kill()
+            .expect("Qemu could not be killed after timeout");
+        panic!("Test timed out after 2 minutes");
+    };
```

Comment:
Rust's default test runner never times out. At worst it shows a warning after 60s. A timeout means that if you suspend the testing for longer than 120s using ctrl-z, you get an unnnecessary test failure. Also depending on how complex the tests of your OS are you may genuinely need more than 120s on low-end hardware. On local systems you can always kill the test manually and on CI there should be a timeout from the CI provider anyway.

Reply:
I don't mind removing this. I don't think the crtl-z case is common, but maybe that just comes down to preference. I never use that myself. 
I mostly added this so I could easily run the tests locally in the background and get notified about failures. 
This bug is random in nature so I ran the tests locally in a loop so I needed some kind of failure state.

As for CI the timeout is 30minutes and that seems excessive. Especially when like here the CPU is maxed out by the hanging test.

> Also depending on how complex the tests of your OS are you may genuinely need more than 120s on low-end hardware.

That might actually explain the failing CI tests. So maybe I should remove this.